contributions/api/controllers/contributions.py

```diff
@@ -125,7 +127,9 @@ def post(token_info):
         try:
             talent_json = in_json["talents"]
             for i in range(len(talent_json)):
+                talent_id = str(uuidlib.uuid4())
```

Comment:
i am wondering, do I need to generate uuid in catalog or not? Since I saw the uuid is generated here

Reply:
yes, I believe that was what we discussed before.

Reply:
Yes, to clarify, the changes in the building block are for accepting IDs and returning IDs in the response. The Talent and Capability ID needs to be generated by the Catalog. Contribution ID, however, is read-only and will be generated by the backend.

Reply:
The line you commented in here is not actually activated in the code since I commented that line out. I think that the current code doesn't have that part because it got commented out with other comments.

Reply:
I see now, thank you both

Reply:
I will modify the code like below. Please check out if this is correct approach.
Ids for talent and capability will be made in the contribution when it get POSTed
For newly added talent/capability during PUT, catalog will add id in its code.
Is it okay? Or let me know if there is other direction.

Reply:
Yes, Talent and Capability ID being a required field needs to be provided by the client (in this case, the Catalog). So, yes, the building block does not have to do anything to add or modify these IDs. Only the Contribution ID (read-only) field needs to be generated by the building block. If you need a quick chat, please let me know. Thanks.

Reply:
Modified based on yesterday's conversation. It is pretty complicated now = )